### PR TITLE
Tidying up the S3 Browser's context menu

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

Clean up of the context menu. Main motivation was to distance more destructive items (delete) from non-destructive items (download)
* added ellipses to the menu items to denote further interactions will happen before the action occurs
* grouped items with separators, and re-ordered them to distance modification based actions from non-modification based actions
* added bucket name to the upload item

A changelog was not added to this PR because this is going into a feature that hasn't been incorporated into mainline yet.

## Testing

Exercised the context menu when selecting 1 item, many items, no items. Verified that the menu item actions fired.

## Screenshots (if appropriate)

WAS:
image

NOW:
image

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties [hunk 1]
```diff
@@ -118,19 +118,19 @@ s3.copy.bucket.action=Copy Name
 s3.copy.bucket.failed=Copy Bucket Failed
 s3.open.viewer.bucket.action=Open Bucket in Viewer
 s3.open.viewer.bucket.failed=Failed to open bucket
-s3.delete.object.action=Delete Object
+s3.delete.object.action=Delete Object...
```
Comment (hunterwerlla) [inline resources/resources/software/aws/toolkits/resources/localized_messages.properties]: I don't like the addition of ellipses, it looks cleaner without them

Comment (hunterwerlla) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: Other than that, looks like a nice UI change!

Comment (rli) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: The UI guidelines say to use ellipsis if the user needs to confirm or provide more input:

https://jetbrains.github.io/ui/controls/button/#10

Comment (hunterwerlla) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: Sounds good, we need to edit other parts our UI to follow this (not as part of this PR, obviously)

**Comment by kiiadi** [comment]
changed text so that it aligns with the AWS console:
Screen Shot 2019-10-24 at 12 39 43 PM

**Comment by kiiadi** [comment]
Removed icon from rename so it's consistent with Rename in IntelliJ Screen Shot 2019-10-24 at 12 54 08 PM

**Pull request merged**